### ISSUE TITLE
failed to pass filename to mpv by config ext-player.js

### ISSUE THREAD

**Issue by ned42** [popcorntime/popcorntime#968]
Operating System Version:
<-- OS X 10.14-->

Popcorn Time Version:
<-- 0.3.10 -->

i saw these code :
if (getPlayerFilenameSwitch(this.get('id')) !== '') {
                // The video file is the biggest file in the torrent
                var videoFile = _.sortBy(streamModel.attributes.torrent.info.files, function (file) {
                    return -file.length;
                })[0];
                cmd += videoFile ? (getPlayerFilenameSwitch(this.get('id')) + "'" + videoFile.name +  "'") : '';
            }

so i add filenameswitch to the mpv like this:
'mpv': {
            type: 'mpv',
            switches: '',
            subswitch: '--sub-file=',
            fs: '--fs',
            filenameswitch: '--title = '
        },
but then when i start streaming the player just didnt cames out,
i also tried to App.vent.trigger the cmd above but nothing shows,so can somebody please explain whats going on?thanks

**Comment by kiriles90** [comment]
I dont think it passes a file to mpv (or the rest of the external players)? It passes a (local) stream.

**Comment by ned42** [comment]
> I dont think it passes a file to mpv (or the rest of the external players)? It passes a (local) stream.

yes i know its a url streaming like host:port/0,but the source code shows that i could get the filename didnt it?

**Comment by kiriles90** [comment]
I see. Are you trying to pass the filename as video title or are you trying to play the file instead of the stream?
If its the first one try this -> diffchecker.com/kkHzgNEs  
There is some weirdness going on with the objects/logic in lines 161-167 which I dont understand. Why do we need to compare file sizes to find the video file since we already have it and can get it more directly? So I changed it abit. (also your filenameswitch: '--title = ' has some extra spaces -> filenameswitch: '--title=')
Let me know if it works because I was testing it with VLC on Windows.

Btw, if someone knows something we are missing about the weird code in l161-7 please explain :)

**Comment by kiriles90** [comment]
I see. Are you trying to pass the filename as video title or are you trying to play the file instead of the stream?
 If its the first one try this -> diffchecker.com/kkHzgNEs
 There is some weirdness going on with the objects/logic in lines 161-167 which I dont understand. Why do we need to compare file sizes to find the video file since we already have it and can get it more directly? So I changed it abit. (also your filenameswitch: '--title = ' has some extra spaces -> filenameswitch: '--title=')
 Let me know if it works because I was testing it with VLC on Windows.

Btw, if someone knows something we are missing about the weird code and apparently bugged code in lines 161-7 please explain :)

**Comment by kiriles90** [comment]
I see. Are you trying to pass the filename as video title or are you trying to play the file instead of the stream?
 If its the first one try this -> diffchecker.com/kkHzgNEs
 There is some weirdness going on with the objects/logic in lines 161-167 which I dont understand. Why do we need to compare file sizes to find the video file since we already have it and can get it more directly? So I changed it abit. (also your filenameswitch: '--title = ' has some extra spaces -> filenameswitch: '--title=')
 Let me know if it works because I was testing it with VLC on Windows.

Btw, if someone knows something we are missing about the weird and apparently bugged code in lines 161-7 please explain :)

**Comment by ned42** [comment]
@kiriles90 thanks for the code,now it can call out the mpv and stream,but the titlename is still like undefined...but i tired fix it as streamModel.get"title",it worked out!
But i still need the filename instead of title cos it better for the subtitile search ... i also tired to access the torrentModel to get info by using export and global and stuff,but i just cant make it right,fuk,,,

**Comment by kiriles90** [comment]
Oops its my fault, forgot to link this diffchecker.com/gcehKHeN ( \src\app\lib\models\stream_info.js) while I had it edited on mine.

**Comment by kiriles90** [comment]
Oops its my fault, forgot to link this diffchecker.com/gcehKHeN ( \src\app\lib\models\stream_info.js) while I had it edited on mine.
If you dont want to edit both files (though I think doing so is more "correct") and you prefer to edit only ext-player.js then -> diffchecker.com/uu2Gbcda

**Comment by kiriles90** [comment]
Oops its my fault, forgot to link diffchecker.com/gcehKHeN ( \src\app\lib\models\stream_info.js) while I had it edited on mine.
If you dont want to edit both files (though I think doing so is more "correct") and you prefer to edit only ext-player.js then just do -> diffchecker.com/uu2Gbcda instead of the two previous diffs.

**Comment by kiriles90** [comment]
Its my fault, forgot to link diffchecker.com/gcehKHeN ( \src\app\lib\models\stream_info.js) while I had it edited on mine.
 If you dont want to edit both files (though I think doing so is more "correct") and you prefer to edit only ext-player.js then just do -> diffchecker.com/acNXjnKU instead of the two previous diffs.

**Comment by kiriles90** [comment]
Its my fault I had an extra edit which made it work and not return undefined. I've edited the comment above with the new code.

**Comment by ned42** [comment]
@kiriles90 but still failed...

**Comment by kiriles90** [comment]
Did you edit everything correctly? Edit the original file not the one you previously edited because I reverted some changes from the diff I linked before. Other than that.. I just tried it on a fresh install and it works (build #279 & VLC but its the same thing, only the command line trigger is different not the function at 161-7). If you're still having trouble link me the error from the console (F12) if you want.

**Comment by kiriles90** [comment]
Did you edit everything correctly? Edit the original file not the one you previously edited because I reverted some changes from the diff I linked before. Other than that.. I just tried it on a fresh install and it works (build 279 & VLC but its the same thing, only the command line trigger is different not the function at 161-7). If you're still having trouble link me the error from the console (F12) if you want.

**Comment by kiriles90** [comment]
Did you edit everything correctly? Edit the original file not the one you previously edited because I reverted some changes from the diff I linked before. Other than that.. I just tried it on a fresh install and it works (build 279 & VLC but its the same thing, only the command line trigger is different not the function at 161-7). If you're still having trouble link me your ext-player.js and the error from the console (F12) if you want.

**Comment by kiriles90** [comment]
Did you edit everything correctly? Edit the original file not the one you previously edited because I reverted some changes from the diff I linked before. Other than that.. I just tried it on a fresh install and it works (with VLC but its the same thing, only the command line trigger is different not the function at 161-7). If you're still having trouble link me your ext-player.js and the error from the console (F12) if you want.
